### PR TITLE
Cat class with two cool tips to choose from!

### PR DESCRIPTION
**Pull request by EOSullivanBerlin**

### CODE REVIEW

File: spec/printer_spec.rb [hunk 1]
```diff
@@ -22,9 +22,9 @@
       expect('output.txt').to have_file_content tips[1]
     end
 
-    it 'speaks the tip(s) when output type is speech' do
-      expect { printer.output(tips, 'speech') }.not_to raise_error # needs changing
-    end
+    # it 'speaks the tip(s) when output type is speech' do
+    #   expect { printer.output(tips, 'speech') }.not_to raise_error # needs changing
+    # end
```
Comment (aballal) [inline spec/printer_spec.rb]: Uncomment and recommit please.

File: test [hunk 2]
```diff
@@ -0,0 +1 @@
+Hello[C
```
Comment (aballal) [inline test]: If this file isn't required delete and commit please.

File: tips/cat.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+b:Did you know that cat -b will tell you the number of non blank lines of output in a file.
+n:The command cat -n will print all lines in a file.
```
Comment (aballal) [inline tips/cat.txt]: The tip doesn't clearly communicate that cat -n prints line numbers against all lines including blank lines and cat -b prints line numbers against all non-blank lines.

File: spec/printer_spec.rb (hunk 1 above)
Comment (aballal) [reply spec/printer_spec.rb]: On second thoughts, please ignore this. It's annoying to do it every single time before committing. We can do it once at the end of the project.

File: spec/cat_spec.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  describe '#suggest_tips' do
+    it 'passes all of the tips if none of the flags are used' do
+      expect(cat.suggest_tips('')).to eq([tips[:b], tips[:n]])
+    end
+
+    describe '-b flag sugestions' do
+      it 'suggests the -b tip when it is not used' do
+        expect(cat.suggest_tips('-n').include?(tips[:b])).to be true
```
Comment (aballal) [inline spec/cat_spec.rb]: Expectations in arguments need to be how the car command would really be used i.e. with a filename. Looks like the way code is written a character of n in filename will break this test. Likewise lots of places.

**Review by aballal — CHANGES_REQUESTED** [review]
Haven't completed the review but feels like it needs a thorough self review and test first.

**Review by aballal — DISMISSED** [review]

**Review by memunyawiri — DISMISSED** [review]
🐑 🇮🇹

**Comment by EOSullivanBerlin** [comment]
Please review!!!!

**Review by BDCraven — DISMISSED** [review]
:shipit:

**Comment by EOSullivanBerlin** [comment]
I need this!!!!

**Review by meta-morpho-sys — DISMISSED** [review]
🐑  IT

**Review by memunyawiri — APPROVED** [review]
:shipit:

**Review by meta-morpho-sys — APPROVED** [review]
shop it!

**Review by BDCraven — APPROVED** [review]
:shipit: !

**Pull request merged**